### PR TITLE
[FLINK-20488][runtime][checkpoint] Show checkpoint type in the Web-UI…

### PR DESCRIPTION
**Pull request by curcur**
… (AC/UC) for each subtask

## What is the purpose of the change
In the UI, show checkpoint type for each subtask (add an unaligned checkpoint column for each subtask).

## Verifying this change

Existing Unit test and also tested on web-UI

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by curcur** [comment]
https://files.slack.com/files-pri/T0763FK1B-F01HMB78WAJ/_____________2020-12-31_______3.53.55.png

File: flink-runtime-web/web-dashboard/src/app/pages/job/checkpoints/subtask/job-checkpoints-subtask.component.html [hunk 1]
```diff
@@ -91,6 +91,7 @@
         <th nzSortKey="alignment.processed" nzShowSort><strong>Processed (persisted) Data</strong></th>
         <th nzSortKey="alignment.duration" nzShowSort><strong>Alignment Duration</strong></th>
         <th nzSortKey="start_delay" nzShowSort><strong>Start Delay</strong></th>
+        <th nzSortKey="unaligned_checkpoint" nzShowSort><strong>Unaligned Checkpoint</strong></th>
```
Comment (rkhachatryan) [inline flink-runtime-web/web-dashboard/src/app/pages/job/checkpoints/subtask/job-checkpoints-subtask.component.html:94]: nit: invert to Aligned to save space?

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java [hunk 2]
```diff
@@ -297,6 +297,11 @@ public void checkpointState(
         try {
             if (takeSnapshotSync(
                     snapshotFutures, metadata, metrics, options, operatorChain, isCanceled)) {
+                LOG.debug(
+                        "Is unaligned checkpoint for task {} : {}",
```
Comment (rkhachatryan) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java:301]: I think a better place to log is the end of `takeSnapshotSync` where other properties are logged.

**Review by rkhachatryan — CHANGES_REQUESTED** [review]
The changes generally LGTM but not to spotless :) Please fix formatting. 
I ran a test job and I think aggregation can be added later as a separate PR.

File: flink-runtime-web/web-dashboard/src/app/pages/job/checkpoints/subtask/job-checkpoints-subtask.component.html (hunk 1 above)
Comment (curcur) [reply flink-runtime-web/web-dashboard/src/app/pages/job/checkpoints/subtask/job-checkpoints-subtask.component.html:94]: you sure...? The meaning of "Aligned" is ambiguous, and I do not think it is straightforward to users that "Aligned" means "not completed as an unaligned checkpoint"?

**Comment by curcur** [comment]
> The changes generally LGTM but not to spotless :) Please fix formatting.
> I ran a test job and I think aggregation can be added later as a separate PR.

hoops, forget to do the formatting ...

File: flink-runtime-web/web-dashboard/src/app/pages/job/checkpoints/subtask/job-checkpoints-subtask.component.html (hunk 1 above)
Comment (rkhachatryan) [reply flink-runtime-web/web-dashboard/src/app/pages/job/checkpoints/subtask/job-checkpoints-subtask.component.html:94]: We could add a tooltip (and it should be documented).
I'm also fine with "Unaligned Checkpoint".

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java (hunk 2 above)
Comment (curcur) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java]: I moved both the log and metrics update to the end of `takeSnapshotSync `.

**Comment by curcur** [comment]
I re-tested after updates.

**Review by rkhachatryan — APPROVED** [review]

**Comment by curcur** [comment]
https://dev.azure.com/mymeiyuan/Flink/_build/results?buildId=152&view=results

local azure is green

the cluster azure fails with unrelated tests

**Review by pnowojski — COMMENTED** [review]
Thanks for the contribution and review @curcur and @rkhachatryan . Merging.

**Pull request merged**